source/getting-started/js-primer.md

```diff
@@ -0,0 +1,154 @@
+}
+```
+
+In the previous example, we can see that the first `console.log(name)` prints out `undefined` instead of the value.
+That is because of a feature of JavaScript called *hosting*.
+Any variable declaration is moved by the programming language to the top of the scope it belongs to.
+As we saw at the beginning, `var` is scoped to the function,
+so the previous example is the same as:
+
+```javascript
+var name;
+console.log(name); // => undefined
+
+if (true) {
+  name = "Tomster";
+
+  console.log(name); // => Tomster
+}
+```
+
+### `const` and `let`
+
+There are two major differences between `var` and both `const` and `let`.
+`const` and `let` are both block-level declarations, and they are *not* hoisted.
+
+Because of this they are not accessible outside of the given block scope (meaning in a `function` or in `{}`) they are declared in.
+You can also not access them before they are declared, or you will get a [`ReferenceError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError).
+
+```javascript
+console.log(name) // => name is not defined
+
+if (person) {
+  console.log(name) // => name is not defined
+  
+  let name = 'Gob Bluth'; // => Gob Bluth
+} else {
+  console.log(name) // => name is not defined
+}
+```
+
+`const` declaration have an additional restriction, they are *constant references*,
+they always refer to the same thing.
+To use a `const` declaration you have to specify the value it refers,
+and you cannot change what the declaration refers to:
+
+```javascript
+const firstName; // => Uncaught SyntaxError: Missing initializer in const declaration
+const firstName = 'Gob';
+firstName = 'George Michael'; // => Uncaught SyntaxError: Identifier 'firstName' has already been declared
+```
+
+Note that `const` does not mean that the value it refers to cannot change.
+If you have an array or an object, you can change their properties:
+
+```javascript
+const myArray = [];
+const myObject = { name: "Tom Dale" };
+
+myArray.push(1);
+myObject.name = "Leah Silber";
+
+console.log(myArray); // => [1]
+console.log(myObject); // => {name: "Leah Silber"}
+```
+
+### `for` loops
+
+Something that might be confusing is the behaviour of `let` in `for` loops.
+
+As we saw before, `let` declarations are scoped to the block they belong to.
+In `for` loops, any variable declared in the for syntax belongs to the loop's block.
+
+Let's look at some code to see what this looks like.
+If you use `var`, this happens:
+
+```javascript
+for (var i = 0; i < 3; i++) {
+  console.log(i) // => wil print 0, 1, 2, and 3 in succession
+}
+
+console.log(i) // => 3
+```
+
+But if you use `let`, this happens instead:
+
+```javascript
+for (let i = 0; i < 3; i++) {
+  console.log(i) // => wil print 0, 1, 2, and 3 in succession
+}
+
+console.log(i) // => i is not defined
+```
+
+Using `let` will avoid accidentally leaking and changing the `i` variable from outside of the `for` block.
```

Comment:
@locks I think we could add a stronger recommendation to prefer `let` and `const` over `var`

Reply:
We need to edit down the guide to do that :P